sa/sa.go

```diff
@@ -498,12 +499,28 @@ func (ssa *SQLStorageAuthority) NewOrderAndAuthzs(ctx context.Context, req *sapb
 		}
 
 		// Second, insert the new order.
-		order := &orderModel{
-			RegistrationID: req.NewOrder.RegistrationID,
-			Expires:        req.NewOrder.Expires.AsTime(),
-			Created:        ssa.clk.Now(),
+		var omv1 orderModelv1
+		var omv2 orderModelv2
+		if features.Get().MultipleCertificateProfiles {
+			omv2 = orderModelv2{
+				RegistrationID:         req.NewOrder.RegistrationID,
+				Expires:                req.NewOrder.Expires.AsTime(),
+				Created:                ssa.clk.Now(),
+				CertificateProfileName: req.NewOrder.CertificateProfileName,
+			}
+		} else {
+			omv1 = orderModelv1{
+				RegistrationID: req.NewOrder.RegistrationID,
+				Expires:        req.NewOrder.Expires.AsTime(),
+				Created:        ssa.clk.Now(),
+			}
+		}
+		var err error
+		if features.Get().MultipleCertificateProfiles {
+			err = tx.Insert(ctx, &omv2)
+		} else {
+			err = tx.Insert(ctx, &omv1)
```

Comment:
This can be collapsed and simplified, something like:

```suggestion
		if features.Get().MultipleCertificateProfiles {
			omv2 := orderModelv2{
				RegistrationID:         req.NewOrder.RegistrationID,
				Expires:                req.NewOrder.Expires.AsTime(),
				Created:                ssa.clk.Now(),
				CertificateProfileName: req.NewOrder.CertificateProfileName,
			}
			err = tx.Insert(ctx, &omv2)
		} else {
			omv1 := orderModelv1{
				RegistrationID: req.NewOrder.RegistrationID,
				Expires:        req.NewOrder.Expires.AsTime(),
				Created:        ssa.clk.Now(),
			}
			err = tx.Insert(ctx, &omv1)
```